frontend/test/pytest/test_operations.py

```diff
@@ -143,5 +145,48 @@ def circuit(x: float, y: float):
     assert np.allclose(qjit_fn(3.14, 0.6), qml_fn(3.14, 0.6))
 
 
+def test_hybrid_op_repr(backend):
+    """Test hybrid operation representation"""
+
+    def circuit(n):
+        quantum_tape = QuantumTape()
+        with qml.QueuingManager.stop_recording(), quantum_tape:
+
+            @for_loop(0, 1, 1)
+            def loop0(_):
+                qml.RX(np.pi, wires=0)
+                return ()
+
+            loop0()
+
+            @while_loop(lambda v: v < 1)
+            def loop1(v):
+                qml.RY(np.pi, wires=0)
+                return v + 1
+
+            loop1(0)
+
+            @cond(n == 1)
+            def cond_fn():
+                qml.RZ(np.pi, wires=0)
+                return 0
+
+            @cond_fn.otherwise
+            def cond_fn():
+                qml.RZ(np.pi, wires=0)
+                return 1
+
+            cond_fn()
+
+        for term in ["ForLoop", "WhileLoop", "Cond", "RX", "RY", "RZ"]:
+            assert term in str(quantum_tape.operations)
+        for op in quantum_tape.operations:
+            if isinstance(op, (ForLoop, WhileLoop, Cond)):
+                assert has_nested_tapes(op)
```

Comment:
It might be good to explicitly test what you are expecting here, so that we don't need to rely on the correctness of `has_nested_tapes` by itself (although it can still be called for verification).

Reply:
I have added the explicit checks, but I kept the original call as well - I am using this test to test the `has_nested_tape` logic. Let me know if you have suggestions regarding this.